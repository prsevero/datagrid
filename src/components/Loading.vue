<template>
    <img src="./loading.svg" />
</template>


<script>
export default {
    name: 'Loading',
};
</script>


<style lang="sass" scoped>
img
    display: block
    margin: 0 auto
    width: 100px
</style>
