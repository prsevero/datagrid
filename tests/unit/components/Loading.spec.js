import { shallowMount } from '@vue/test-utils';
import Loading from '@/components/Loading.vue';

describe('Loading.vue', () => {

    let props;
    let mountedLoading;
    const loading = () => {
        if (!mountedLoading) {
            mountedLoading = shallowMount(Loading, {propsData: props});
        }
        return mountedLoading;
    }

    beforeEach(() => {
        mountedLoading = undefined;
    });


    // All tests will go here
    it('should match snapshot', () => {
        const wrapper = loading();
        expect(wrapper.vm.$el).toMatchSnapshot();
    });

    it('always render an img', () => {
        const wrapper = loading();
        expect(wrapper.is('img')).toBe(true);
    });

});
