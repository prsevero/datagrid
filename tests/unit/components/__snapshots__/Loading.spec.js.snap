// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading.vue should match snapshot 1`] = `
<img
  src="./loading.svg"
/>
`;
